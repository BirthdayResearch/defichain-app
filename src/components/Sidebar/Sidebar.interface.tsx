export interface SidebarProps {}

export interface SidebarState {
  balance: {
    available: string;
  };
}
