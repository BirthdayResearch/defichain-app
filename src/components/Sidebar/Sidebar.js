import React, { Component } from 'react'
import { ReactSVG } from 'react-svg'
import logo from '../../assets/svg/logo-defi.svg';
import {
  Nav,
  NavItem,
  NavLink
} from 'reactstrap';
import { NavLink as RRNavLink, withRouter } from 'react-router-dom';
import {
  MdAccountBalanceWallet,
  MdDns,
  MdViewWeek,
  MdCompareArrows
} from "react-icons/md";
import styles from './Sidebar.module.scss';

class Sidebar extends Component {
  render() {
    return (
      <div className={styles.sidebar}>
        <div className={styles.brand}>
          <ReactSVG src={logo} />
        </div>
        <div className={styles.navs}>
          <Nav className={`${styles.navMain} flex-column nav-pills`}>
            <NavItem className={styles.navItem}>
              <NavLink
                to="/" exact
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
                isActive={(match, location) => {
                  return location.pathname.startsWith("/wallet") || location.pathname === "/";
                }}
              >
                <MdAccountBalanceWallet />
                Wallet
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink to="/masternodes" tag={RRNavLink} className={styles.navLink} activeClassName={styles.active}>
                <MdDns />
                Masternodes
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink to="/blockchain" tag={RRNavLink} className={styles.navLink} activeClassName={styles.active}>
                <MdViewWeek />
                Blockchain
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink to="/exchange" tag={RRNavLink} className={styles.navLink} activeClassName={styles.active}>
                <MdCompareArrows />
                Exchange
              </NavLink>
            </NavItem>
          </Nav>
          <Nav className={`${styles.navSub} flex-column nav-pills`}>
            <NavItem className={styles.navItem}>
              <NavLink to="/help" tag={RRNavLink} className={styles.navLink} activeClassName={styles.active}>
                Help
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink to="/settings" tag={RRNavLink} className={styles.navLink} activeClassName={styles.active}>
                Settings
              </NavLink>
            </NavItem>
          </Nav>
        </div>
      </div>
    );
  }
}

export default withRouter(Sidebar);