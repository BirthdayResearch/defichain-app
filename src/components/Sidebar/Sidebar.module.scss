// App and Bootstrap variables required
@import '~bootstrap/scss/functions';
@import "~bootstrap/scss/variables";
@import "~bootstrap/scss/mixins";
@import '../../scss/variables';
@import '../../scss/bootstrap/variables';


.sidebar {
  width: 224px;
  background-color: $white;
  display: flex;
  flex-direction: column;
  flex: 0 0 224px;
  z-index: 2;
}

.balance {
  position: relative;
  height: 112px;
  border-bottom: $border-width solid $border-color;
  padding: 28px 24px 20px;
  display: flex;
  flex-shrink: 0;
  flex-wrap: wrap;
  align-items: flex-end;
  align-content: flex-end;

  &:before {
    content: "";
    position: absolute;
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    z-index: -1;
    -webkit-app-region: drag;
  }
}

.balanceLabel,
.balanceValue {
  flex-basis: 100%;
  font-size: .875rem;
  line-height: 1.25rem;
}

.balanceLabel {
  font-weight: $font-weight-semibold;
}

.balanceValue {
  color: $secondary;
}

.navs {
  flex-grow: 1;
  padding: 16px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.navMain,
.navSub {
  margin-top: 12px;
  margin-bottom: 16px;

  .navItem {
    margin-top: 4px;
    margin-bottom: 4px;
  }

  .navLink {
    color: $secondary;
    display: flex;
    align-items: center;
    padding-top: .75rem;
    padding-bottom: .75rem;
    text-transform: none;
    font-weight: $font-weight-normal;

    svg {
      width: 24px;
      height: 24px;
      margin-right: 12px;
    }
  }

  .active {
    background-color: $dfi-scent;
    color: $body-color;

    svg {
      color: $primary;
    }
  }
}