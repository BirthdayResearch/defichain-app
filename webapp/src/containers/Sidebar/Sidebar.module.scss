// App and Bootstrap variables required
@import '../../scss/bootstrap/common';

.sidebar {
  width: 224px;
  background-color: $white;
  display: flex;
  flex-direction: column;
  flex: 0 0 224px;
  z-index: 2;
}

.blur {
  background: $gray-200;
}

div[disabled] {
  pointer-events: none;
  opacity: 0.7;
}

.iconSideBar {
  min-height: 26px;
}

.iconNavTop {
  color: $secondary;
  cursor: pointer;
  &:hover:not(.navBadgeLocked) {
    color: $black;
    transition: all 0.2ms linear;
  }
}

.currentNetwork {
  position: relative;
  height: 70px;
  border-bottom: $border-width solid $border-color;
  padding: 0px 20px 25px;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;

  &:before {
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    z-index: -1;
    -webkit-app-region: drag;
  }
}

.currentNetworkHeading {
  font-size: $font-size-sm;
  font-weight: $font-weight-medium;
}

.currentNetworkValue {
  display: flex;
  font-size: $font-size-sm;
  color: $secondary;
}

.balanceLabel,
.balanceValue {
  flex-basis: 100%;
  font-size: 0.875rem;
  line-height: 1.25rem;
}

.balanceLabel {
  font-weight: $font-weight-semibold;
}

.balanceValue {
  color: $secondary;
}

.navs {
  flex-grow: 1;
  padding: 16px;
  min-height: 384px;
  display: flex;
  flex-wrap: nowrap;
  flex-direction: column;
  justify-content: space-between;
  overflow: auto;
}

.navMain,
.navSub {
  margin-top: 12px;
  margin-bottom: 16px;
  flex-wrap: nowrap;

  .navItem {
    margin: 0.25rem 0;
  }

  .navLink {
    color: $secondary;
    display: flex;
    align-items: center;
    padding: 0.75rem 1rem;

    text-transform: none;
    font-weight: $font-weight-normal;

    border: none;
    border-radius: 0.25rem;

    transition: color, background-color 250ms;

    &:hover:not(.active) {
      color: inherit;
    }

    svg {
      width: 24px;
      height: 24px;
      margin-right: 12px;
    }
  }

  .active {
    background-color: $dfi-scent;
    color: $body-color;

    svg {
      color: $primary;
    }
  }
}

.navSubIcon {
  margin: 0;
  justify-content: center;

  .navLink {
    padding: 0.75rem;
  }

  svg {
    margin-right: 0 !important;
  }
}

.navBadgeIcon {
  background-color: $light;
  font-size: 1.5rem;
  color: $secondary;
  padding: 0.75rem;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 1.5rem;
  &:hover:not(.active) {
    background-color: $dfi-scent;
    color: $primary;
  }
}

.navBadgeText {
  text-align: center;
  color: $secondary;
  font-weight: 500;
  font-size: 0.95rem;
}

.passphraseIcon {
  padding: 0.75rem 16px 0.25rem;
}

.iconPosition {
  left: 12px;
  position: relative;
}

.navBadgeLocked {
  color: $primary;
}
