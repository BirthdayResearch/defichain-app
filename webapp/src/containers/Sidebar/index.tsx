import React, { useEffect, useState } from 'react';
import isEmpty from 'lodash/isEmpty';
import { Nav, NavItem, NavLink } from 'reactstrap';
import {
  NavLink as RRNavLink,
  withRouter,
  RouteComponentProps,
} from 'react-router-dom';
import { I18n } from 'react-redux-i18n';
import { connect } from 'react-redux';
import {
  MdAccountBalanceWallet,
  MdDns,
  MdViewWeek,
  MdToll,
  MdPieChart,
  MdSettings,
  MdHelp,
  MdExitToApp,
  MdLockOutline,
  MdLockOpen,
  MdSwapVert,
  MdSwapHoriz,
} from 'react-icons/md';
import { HiTerminal } from 'react-icons/hi';
import {
  fetchInstantBalanceRequest,
  lockWalletStart,
} from '../WalletPage/reducer';
import SyncStatus from '../SyncStatus';
import {
  BLOCKCHAIN_BASE_PATH,
  CONSOLE_RPC_CALL_BASE_PATH,
  WALLET_PAGE_PATH,
  WALLET_BASE_PATH,
  MASTER_NODES_PATH,
  TOKENS_PATH,
  SWAP_PATH,
  WALLET_TOKENS_PATH,
  LIQUIDITY_PATH,
  HELP_PATH,
  SETTING_PATH,
  PACKAGE_VERSION,
  TRASACTIONS_PATH,
} from '../../constants';
import styles from './Sidebar.module.scss';
import { updateBalanceScheduler } from '../../worker/schedular';
import usePrevious from '../../components/UsePrevious';
import {
  openEncryptWalletModal,
  openWalletPassphraseModal,
  openExitWalletModal,
  openMasternodeWarningModal,
} from '../PopOver/reducer';
import { MasterNodeObject } from '../MasternodesPage/masterNodeInterface';
import { hasAnyMasternodeEnabled } from '../MasternodesPage/service';

export interface SidebarProps extends RouteComponentProps {
  fetchInstantBalanceRequest: () => void;
  walletBalance: string;
  unit: string;
  isErrorModalOpen: boolean;
  blockChainInfo: any;
  isWalletUnlocked: boolean;
  isLoadingRemovePoolLiquidity: boolean;
  isLoadingAddPoolLiquidity: boolean;
  isLoadingPoolSwap: boolean;
  openEncryptWalletModal: () => void;
  openWalletPassphraseModal: () => void;
  lockWalletStart: () => void;
  isWalletCreatedFlag: boolean;
  openExitWalletModal: (t: boolean) => void;
  isWalletEncrypted: boolean;
  myMasternodes: MasterNodeObject[];
  openMasternodeWarningModal: (t: boolean) => void;
}

const Sidebar: React.FunctionComponent<SidebarProps> = (props) => {
  const prevIsErrorModalOpen = usePrevious(props.isErrorModalOpen);
  const [blur, setBlur] = useState(true);

  useEffect(() => {
    props.fetchInstantBalanceRequest();
    const clearWalletBalanceTimer = updateBalanceScheduler();
    return () => {
      clearWalletBalanceTimer();
    };
  }, []);

  useEffect(() => {
    if (!props.isErrorModalOpen && prevIsErrorModalOpen) {
      props.fetchInstantBalanceRequest();
    }
  }, [prevIsErrorModalOpen, props.isErrorModalOpen]);

  const {
    openEncryptWalletModal,
    openWalletPassphraseModal,
    isWalletUnlocked,
    lockWalletStart,
    isLoadingRemovePoolLiquidity,
    isLoadingAddPoolLiquidity,
    isLoadingPoolSwap,
    isWalletCreatedFlag,
    openExitWalletModal,
    blockChainInfo,
    isWalletEncrypted,
    myMasternodes,
    openMasternodeWarningModal,
  } = props;

  useEffect(() => {
    setBlur(!blur);
  }, [
    isLoadingRemovePoolLiquidity,
    isLoadingAddPoolLiquidity,
    isLoadingPoolSwap,
  ]);

  const chainName = !isEmpty(blockChainInfo)
    ? blockChainInfo.chain.charAt(0).toUpperCase() +
      blockChainInfo.chain.slice(1)
    : '';

  const ICON_SIZE = 25;

  const onLockWallet = () => {
    hasAnyMasternodeEnabled(myMasternodes)
      ? openMasternodeWarningModal(true)
      : lockWalletStart();
  };

  return (
    <div className={`${styles.sidebar} ${blur && styles.blur}`} disabled={blur}>
      <div className={`text-right m-2 mr-3 ${styles.iconSideBar}`}>
        {isWalletCreatedFlag && (
          <>
            {!isWalletEncrypted ? (
              <MdLockOpen
                className={`${styles.iconNavTop}`}
                size={ICON_SIZE}
                onClick={openEncryptWalletModal}
                title={I18n.t('containers.wallet.encryptWalletPage.title')}
              />
            ) : isWalletUnlocked ? (
              <MdLockOpen
                className={`${styles.iconNavTop}`}
                size={ICON_SIZE}
                onClick={onLockWallet}
                title={I18n.t('alerts.encryptWalletNoticeTitle')}
              />
            ) : (
              <MdLockOutline
                className={`${styles.iconNavTop} ${styles.navBadgeLocked}`}
                size={ICON_SIZE}
                onClick={openWalletPassphraseModal}
                title={I18n.t('containers.wallet.unlockWalletPage.title')}
              />
            )}
            <MdExitToApp
              onClick={() => openExitWalletModal(true)}
              className={`ml-2 ${styles.flipX} ${styles.iconNavTop}`}
              size={ICON_SIZE}
              title={I18n.t('alerts.exitWallet')}
            />
          </>
        )}
      </div>
      <div className={styles.currentNetwork}>
        <div className={styles.currentNetworkHeading}>
          {I18n.t('components.syncStatus.network')}
        </div>
        <div className={styles.currentNetworkValue}>
          <div>{chainName}</div>
          <div className='ml-auto'>{PACKAGE_VERSION}</div>
        </div>
      </div>
      <div className={styles.navs}>
        {!isWalletCreatedFlag ? (
          <Nav className={`align-items-center flex-grow-1`}>
            <NavItem>
              <NavLink
                className={`d-flex align-items-center justify-content-center flex-column p-0`}
                to={WALLET_TOKENS_PATH}
                exact
                tag={RRNavLink}
              >
                <div className={`${styles.navBadgeIcon} rounded-circle`}>
                  <MdAccountBalanceWallet />
                </div>
                <div className={`${styles.navBadgeText}`}>
                  {I18n.t('containers.sideBar.selectCreateRestore')}
                </div>
              </NavLink>
            </NavItem>
          </Nav>
        ) : (
          <Nav className={`${styles.navMain} flex-column nav-pills`}>
            <NavItem className={styles.navItem}>
              <NavLink
                to={WALLET_TOKENS_PATH}
                exact
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
                isActive={(_match: any, location: { pathname: string }) => {
                  return (
                    location.pathname.startsWith(WALLET_BASE_PATH) ||
                    location.pathname === WALLET_PAGE_PATH
                  );
                }}
              >
                <MdAccountBalanceWallet />
                {I18n.t('containers.sideBar.balances')}
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink
                to={TRASACTIONS_PATH}
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                <MdSwapVert />
                {I18n.t('containers.sideBar.transactions')}
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink
                to={LIQUIDITY_PATH}
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                <MdPieChart />
                {I18n.t('containers.sideBar.liquidity')}
              </NavLink>
            </NavItem>

            <NavItem className={styles.navItem}>
              <NavLink
                to={SWAP_PATH}
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                <MdSwapHoriz />
                {I18n.t('containers.sideBar.dex')}
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink
                to={TOKENS_PATH}
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                <MdToll />
                {I18n.t('containers.sideBar.tokens')}
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink
                to={BLOCKCHAIN_BASE_PATH}
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                <MdViewWeek />
                {I18n.t('containers.sideBar.blockchain')}
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink
                to={MASTER_NODES_PATH}
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                <MdDns />
                {I18n.t('containers.sideBar.masterNodes')}
                {hasAnyMasternodeEnabled(myMasternodes) && (
                  <div className={styles.iconPosition}>
                    <span className={`txn-status-enabled mt-1 ml-1`}></span>
                  </div>
                )}
              </NavLink>
            </NavItem>
          </Nav>
        )}
        <Nav className={`${styles.navSub} ${styles.navSubIcon}`}>
          <NavItem>
            <NavLink
              to={CONSOLE_RPC_CALL_BASE_PATH}
              tag={RRNavLink}
              className={styles.navLink}
              activeClassName={styles.active}
              title={I18n.t('containers.sideBar.console')}
            >
              <HiTerminal />
            </NavLink>
          </NavItem>

          <NavItem>
            <NavLink
              to={HELP_PATH}
              tag={RRNavLink}
              className={styles.navLink}
              activeClassName={styles.active}
              title={I18n.t('containers.sideBar.help')}
            >
              <MdHelp />
            </NavLink>
          </NavItem>

          <NavItem>
            <NavLink
              to={SETTING_PATH}
              tag={RRNavLink}
              className={styles.navLink}
              activeClassName={styles.active}
              title={I18n.t('containers.sideBar.settings')}
            >
              <MdSettings />
            </NavLink>
          </NavItem>
        </Nav>
      </div>
      <SyncStatus />
    </div>
  );
};

const mapStateToProps = (state) => {
  const { i18n, wallet, settings, popover, swap, masterNodes } = state;
  return {
    locale: i18n.locale,
    unit: settings.appConfig.unit,
    walletBalance: wallet.walletBalance,
    isErrorModalOpen: popover.isOpen,
    blockChainInfo: wallet.blockChainInfo,
    isWalletUnlocked: wallet.isWalletUnlocked,
    isLoadingRemovePoolLiquidity: swap.isLoadingRemovePoolLiquidity,
    isLoadingAddPoolLiquidity: swap.isLoadingAddPoolLiquidity,
    isLoadingPoolSwap: swap.isLoadingPoolSwap,
    isWalletCreatedFlag: wallet.isWalletCreatedFlag,
    isWalletEncrypted: wallet.isWalletEncrypted,
    myMasternodes: masterNodes.myMasternodes,
  };
};

const mapDispatchToProps = {
  fetchInstantBalanceRequest,
  openEncryptWalletModal,
  openWalletPassphraseModal,
  lockWalletStart,
  openExitWalletModal,
  openMasternodeWarningModal,
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Sidebar)
);
