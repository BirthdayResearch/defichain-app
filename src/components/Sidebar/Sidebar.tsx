import React, { Component } from "react";
import { Nav, NavItem, NavLink } from "reactstrap";
import {
  NavLink as RRNavLink,
  withRouter,
  RouteComponentProps
} from "react-router-dom";
import {
  MdAccountBalanceWallet,
  MdDns,
  MdViewWeek,
  MdCompareArrows
} from "react-icons/md";
import styles from "./Sidebar.module.scss";
import { SidebarProps, SidebarState } from "./Sidebar.interface";

class Sidebar extends Component<
  SidebarProps & RouteComponentProps,
  SidebarState
> {
  state = {
    balance: {
      available: "1,000"
    }
  };

  render() {
    return (
      <div className={styles.sidebar}>
        <div className={styles.balance}>
          <div className={styles.balanceLabel}>Balance</div>
          <div className={styles.balanceValue}>
            {this.state.balance.available} DFI
          </div>
        </div>
        <div className={styles.navs}>
          <Nav className={`${styles.navMain} flex-column nav-pills`}>
            <NavItem className={styles.navItem}>
              <NavLink
                to="/"
                exact
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
                isActive={(match, location) => {
                  return (
                    location.pathname.startsWith("/wallet") ||
                    location.pathname === "/"
                  );
                }}
              >
                <MdAccountBalanceWallet />
                Wallet
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink
                to="/masternodes"
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                <MdDns />
                Masternodes
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink
                to="/blockchain"
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                <MdViewWeek />
                Blockchain
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink
                to="/exchange"
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                <MdCompareArrows />
                Exchange
              </NavLink>
            </NavItem>
          </Nav>
          <Nav className={`${styles.navSub} flex-column nav-pills`}>
            <NavItem className={styles.navItem}>
              <NavLink
                to="/help"
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                Help
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink
                to="/settings"
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                Settings
              </NavLink>
            </NavItem>
          </Nav>
        </div>
      </div>
    );
  }
}

export default withRouter(Sidebar);
