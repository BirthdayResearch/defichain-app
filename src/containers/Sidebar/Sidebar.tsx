import React, { Component } from "react";
import { Nav, NavItem, NavLink } from "reactstrap";
import {
  NavLink as RRNavLink,
  withRouter,
  RouteComponentProps,
} from "react-router-dom";
import { I18n } from "react-redux-i18n";
import {
  MdAccountBalanceWallet,
  MdDns,
  MdViewWeek,
  MdCompareArrows,
} from "react-icons/md";
import styles from "./Sidebar.module.scss";
import { SidebarProps, SidebarState } from "./Sidebar.interface";
import { connect } from "react-redux";

class Sidebar extends Component<
  SidebarProps & RouteComponentProps,
  SidebarState
> {
  state = {
    balance: {
      available: "1,000",
    },
  };

  render() {
    return (
      <div className={styles.sidebar}>
        <div className={styles.balance}>
          <div className={styles.balanceLabel}>
            {I18n.t("containers.sideBar.balance")}
          </div>
          <div className={styles.balanceValue}>
            {this.state.balance.available} {I18n.t("containers.sideBar.dfi")}
          </div>
        </div>
        <div className={styles.navs}>
          <Nav className={`${styles.navMain} flex-column nav-pills`}>
            <NavItem className={styles.navItem}>
              <NavLink
                to="/"
                exact
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
                isActive={(match, location) => {
                  return (
                    location.pathname.startsWith("/wallet") ||
                    location.pathname === "/"
                  );
                }}
              >
                <MdAccountBalanceWallet />
                {I18n.t("containers.sideBar.wallet")}
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink
                to="/masternodes"
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                <MdDns />
                {I18n.t("containers.sideBar.masterNodes")}
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink
                to="/blockchain"
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                <MdViewWeek />
                {I18n.t("containers.sideBar.blockchain")}
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink
                to="/exchange"
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                <MdCompareArrows />
                {I18n.t("containers.sideBar.exchange")}
              </NavLink>
            </NavItem>
          </Nav>
          <Nav className={`${styles.navSub} flex-column nav-pills`}>
            <NavItem className={styles.navItem}>
              <NavLink
                to="/help"
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                {I18n.t("containers.sideBar.help")}
              </NavLink>
            </NavItem>
            <NavItem className={styles.navItem}>
              <NavLink
                to="/settings"
                tag={RRNavLink}
                className={styles.navLink}
                activeClassName={styles.active}
              >
                {I18n.t("containers.sideBar.settings")}
              </NavLink>
            </NavItem>
          </Nav>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const { locale } = state.i18n;
  return {
    locale,
  };
};

export default withRouter(connect(mapStateToProps)(Sidebar));
