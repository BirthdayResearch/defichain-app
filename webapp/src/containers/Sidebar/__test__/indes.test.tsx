import React from 'react';
import { shallow } from 'enzyme';
import Sidebar from '../index';

describe('Sidebar Page', () => {
  it('should check for snapshot', () => {
    const wrapper = shallow(<Sidebar />);
    expect(wrapper).toMatchSnapshot();
  });
});
