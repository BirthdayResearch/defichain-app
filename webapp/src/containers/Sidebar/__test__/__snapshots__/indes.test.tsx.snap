// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Page should check for snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
